src/application.py

```diff
@@ -1,9 +1,101 @@
-from fastapi import FastAPI
+from fastapi import FastAPI, HTTPException
 from fastapi.responses import PlainTextResponse
+from pydantic import PositiveInt, EmailStr
+from pydantic_extra_types.country import CountryAlpha2
+
+from src.models import User, Activity, SuperUser, SuperUserRoles, ActivityTypes
+from tools.tools import short_uuid4_generator
 
 app = FastAPI()
 
 
 @app.get("/", response_class=PlainTextResponse)
 async def root():
     return "Hello, I'm good!"
+
+
+emails_db = []
+users_db = []
+superusers_db = []
+activities_db = []
+users_id_db = []
+superusers_id_db = []
+activities_id_db = []
+
+
+@app.post("/users/", response_model=User)
+def post_user(
+    username: str,
+    email: EmailStr,
+    age: PositiveInt = None,
+    country: CountryAlpha2 = None,
+) -> User:
+    user_id = short_uuid4_generator()
+    user = User(
+        user_id=user_id, username=username, email=email, age=age, country=country
+    )
+    if user.email in emails_db:
+        raise HTTPException(status_code=400, detail="Email already registered")
+    users_id_db.append(user.user_id)
+    users_db.append(user)
+    emails_db.append(user.email)
+    return user
+
+
+@app.post("/superusers/", response_model=SuperUser)
+def post_superuser(
+    user_id: PositiveInt,
```

Comment:
Since SuperUser is also a user, superuser id should also be unique and generated by our app, and also stored in the user_id_db.

Reply:
Alright!

Reply:
you still have the unused user_id as the argument in post :)